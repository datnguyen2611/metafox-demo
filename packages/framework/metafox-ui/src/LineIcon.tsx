import { RefOf } from '@metafox/framework';
import { LineIconProps } from '@metafox/ui';
import { styled } from '@mui/material';
import clsx from 'clsx';
import React from 'react';
import './lineicon.css';

const regexPattern = /^ico-/;

const StyledIcon = styled('span', {
  name: 'MuiIcon',
  slot: 'root',
  skipSx: false,
  overridesResolver: (props, styles) => [styles.root]
})({
  fontFamily: "'lineficon' !important",
  speak: 'none',
  fontStyle: 'normal',
  fontWeight: 'normal',
  fontVariant: 'normal',
  textTransform: 'none',
  fontSmoothing: 'antialiased',
  WebkitFontSmoothing: 'antialiased'
  // lineHeight: 'inherit'
});

export default React.forwardRef(
  (
    {
      component: AsComponent = 'span',
      icon = 'ico-repeat-alt',
      className,
      ...props
    }: LineIconProps,
    ref: RefOf<HTMLSpanElement>
  ) => {
    // support with prefix Mobile App icon (old data)
    const iconExtra =
      icon && regexPattern.test(icon) ? undefined : `ico-${icon}`;

    return (
      <StyledIcon
        ref={ref}
        className={clsx(className, 'ico', icon, iconExtra)}
        {...props}
      />
    );
  }
);
