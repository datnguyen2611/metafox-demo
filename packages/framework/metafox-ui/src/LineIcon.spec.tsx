import { render } from '@testing-library/react';
import * as React from 'react';
import LineIcon from './LineIcon';

describe('LineIcon', () => {
  it('render icon', async () => {
    const { container } = render(<LineIcon icon="ico-angle-right" />);

    expect(container).toMatchSnapshot();
  });
});
