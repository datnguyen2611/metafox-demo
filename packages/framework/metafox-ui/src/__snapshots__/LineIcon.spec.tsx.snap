// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineIcon render icon 1`] = `
<div>
  <span
    class="ico ico-angle-right css-1o2x2pn-MuiIcon-root"
  />
</div>
`;
